### ISSUE TITLE
Switching to another app and back may make Osmand lose context

### ISSUE THREAD

**Issue by speleoluc** [osmandapp/OsmAnd#20012]
### Description

When another app is brought to the forefront and then Osmand returns to the forefront, it has often (not always) "lost context":, i.e. the displayed track may have been discarded and needs to be reopened, the selected profile may have changed to the generic one (Earth icon), the map style may be returned to default (instead of nautical in my case), the UI language may be changed to OS default (my OS default is French, but I normally run Osmand+ in English) or the display is back to the splash screen and the map tiles get reloaded one at a time.

Could this be a memory problem? I still have 664 Mb available to apps.

### Steps to reproduce

Open an existing track in Osmand+, bring up another app to the forefront and return to Osmand+. Sometimes, the track is no longer displayed and other UI changes have occurred. As any intermittent bug, it doesn't occur when you try to reproduce it...

### Actual result

Osmand+ doesn't re-apper as it was left.

### Expected result

Osmand+ should re-apper as it was left.

### Your Environment (required)

OsmAnd Version:4.7.17
Android version: 8.0.0
Device model: Samsung A5 (2017) phone
Crash-Logs: no crash

**Comment by yuriiurshuliak** [comment]
The bug has been reproduced. To reproduce it, I opened a track, selected a boat profile, and calculated the navigation for that track. Then, I switched to another program and kept it in the foreground for about a minute. Upon returning to the OsmAnd, the track was missing, the navigation calculation was gone, and the profile had changed. Power saving mode was not enabled.

OsmAnd~ 4.8.0#2454m, released: 2024-06-07


https://github.com/osmandapp/OsmAnd/assets/127092082/ffcc70c9-ef54-4fc5-a141-3ad09edbf992

**Comment by speleoluc** [comment]
Has it been assigned to a future milestone, or will the profile handling remain buggy and thus, somewhat useless?

**Comment by vshcherb** [comment]
Yes now we have more critical issues as restoring navigation after application was killed. On many modern devices app is not killed instantly
